### PR TITLE
adding csm models

### PR DESCRIPTION
**Pull request by villrv**
Added the Chatzopoulos, Wheeler & Vinko 2011 CSM models for s=0 and s=2

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ script:
     - mpirun -np 2 coverage run -p --source=mosfit -m mosfit -e SN2006le --travis -i 1 -f 1 -p 0
     - mpirun -np 2 coverage run -p --source=mosfit -m mosfit -e SN2006le.json --travis -i 1 --no-fracking -m magnetar
     - mpirun -np 2 coverage run -p --source=mosfit -m mosfit -e LSQ12dlf --travis -i 1 --no-fracking -m slsn
+    -mpirun -np 2 coverage run -p --source=mostfit -m mostfit -e LSQ12dlf --travis -i 1 --no-fracking -m csm
```
Comment (guillochon) [inline .travis.yml]: Missing space (should be `- mpirun`) and `mostfit` should be spelled `mosfit`.

File: mosfit/modules/engines/csm.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import numpy as np
+from scipy import interpolate
+from mosfit.modules.engines.engine import Engine
+from math import isnan
+
+CLASS_NAME = 'CSM'
+
+class CSM(Engine):
+    """Magnetar spin-down engine
```
Comment (guillochon) [inline mosfit/modules/engines/csm.py]: Should update Docstrings, provide citation to Manos' paper

**Comment by villrv** [comment]
updated typo and added citations

File: mosfit/modules/engines/csm.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        super().__init__(**kwargs)
+
+    def process(self, **kwargs):
+        self._s = kwargs['s'] #preset in the model name for now
+        self._delta = kwargs['delta'] # [0,3)
+        self._n = kwargs['n'] # [6,10]
+        self._kappa = kwargs['kappa']
+        self._R0 = kwargs['r0'] * 1.496e13 #AU to cm
+        self._mejecta = kwargs['mejecta'] * 1.9884e33#Msol to grms
+        self._mnickel = kwargs['mejecta'] * kwargs['fnickel']
```
Comment (guillochon) [inline mosfit/modules/engines/csm.py]: This line isn't needed since you don't have any Nickel in this model

File: models/csm/csm.json [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    "redshift":{
+        "kind":"parameter",
+        "value":0.0,
+        "class":"parameter",
+        "inputs":[
+            "transient"
+        ],
+        "latex":"z"
+    },
+    "fnickel":{
```
Comment (guillochon) [inline models/csm/csm.json]: fnickel can be removed as a parameter since it's not used.

File: models/csm/parameters.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        "max_value":7.0
+    },
+    "vejecta":{
+        "min_value":1.0e4,
+        "max_value":1.0e4,
+        "log":true
+    },
+    "rho":{
+        "min_value":-13.0,
+        "max_value":-11.0
```
Comment (guillochon) [inline models/csm/parameters.json]: Should add a `"log":true` to this?

File: models/csm/parameters.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
+{
+    "nhhost":{
+        "value":0.0
+    },
+    "fnickel":{
```
Comment (guillochon) [inline models/csm/parameters.json]: Can remove `fnickel`

File: models/csm/parameters.json [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        "max_value":100.0
+    },
+    "mcsm":{
+        "min_value":0.1,
+        "max_value":2.0
+    },
+    "texplosion":{
+        "value":0.0
+    },
+    "temperature":{
```
Comment (guillochon) [inline models/csm/parameters.json]: This variable isn't used and can be removed.

**Pull request merged**